### PR TITLE
[IMP] pylint_pr.cfg: Add new checks from pylint-odoo

### PR DESCRIPTION
**Pull request by moylop260**
Enable next pr and beta checks
 -  api-one-deprecated - W8104 - Now is a decorator deprecated see more info [here](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#field)
 -  attribute-deprecated - W8105 - _fields and _defaults Except in odoo 6.1
 -  javascript-lint - W7903 - Use of jshint script.
 -  license-allowed - C8105 - AGPL-3,GPL-2,GPL-2 or any later version,GPL-3,GPL-3 or any later version,LGPL-3,Other OSI approved licence,Other proprietary (all from [odoo] (https://github.com/odoo/odoo/blob/57168a9e90066ed6b3d274d5032a20bef342a00f/openerp/addons/base/module/module.py#L283-L292))
 -  method-required-super - W8106 - Next method require a `super` call create,write,read,unlink,copy,setUp,tearDown,default_get (Extracted with `rgrep -B10 super . --include=*.py` in official modules)

The configuration of this PR is:

| msg  | 9.0  | 8.0  | 7.0  | 6.1  | is beta msg?    |   PR or branches  |
|---|---|---|---|---|---|---|
|   api-one-deprecated  |   yes |   yes |   n/a |   n/a |    [yes] (https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_beta.cfg#L30-L52)  |   [PR](https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_pr.cfg#L48-L70)  |
|   Attribute-deprecated    |   yes |   yes |   [no](https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_exclude_70.cfg#L18)  |  [no](https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_exclude_61.cfg#L18)  |   [yes] (https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_beta.cfg#L30-L52) |   [PR](https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_pr.cfg#L48-L70)  |
|   Javascript-lint |   yes |   yes |   yes |   n/a |    [yes] (https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_beta.cfg#L30-L52)  |   [PR](https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_pr.cfg#L48-L70)  |
|   license-allowed |   yes |   yes |   yes |   yes |    [yes] (https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_beta.cfg#L30-L52)   |   [PR](https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_pr.cfg#L48-L70)  |
|   method-required-super   |   yes |   yes |   yes |   yes |    [yes] (https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_beta.cfg#L30-L52) |   [PR](https://github.com/vauxoo-dev/maintainer-quality-tools/blob/oca-master-enables/travis/cfg/travis_run_pylint_pr.cfg#L48-L70)  |

### CODE REVIEW

File: travis/cfg/travis_run_pylint_beta.cfg [hunk 1]
```diff
@@ -2,16 +2,21 @@
 #  pylint configuration file will display them without affecting your build status.
 
 # Beta message and code:
+#   api-one-deprecated - W8104
```
Comment (dreispt) [inline travis/cfg/travis_run_pylint_beta.cfg]: This one is for >= 9.0
IMO it's work mentioning that in a comment

File: travis/cfg/travis_run_pylint_pr.cfg [hunk 2]
```diff
@@ -13,21 +13,29 @@ readme_template_url="https://github.com/OCA/maintainer-tools/blob/master/templat
 manifest_required_author="Odoo Community Association (OCA)"
 manifest_required_keys=license
 manifest_deprecated_keys=description,active
+attribute_deprecated=_fields,_defaults
```
Comment (dreispt) [inline travis/cfg/travis_run_pylint_pr.cfg]: This generates PR build errors, correct? 
It should apply only to >= 8.0

File: travis/cfg/travis_run_pylint_beta.cfg (hunk 1 above)
Comment (moylop260) [inline travis/cfg/travis_run_pylint_beta.cfg]: If you use `api.one` in v6 or v7 you will have server errors and pylint fails too.
If you use `api.one` in v8 or v9 you will have just pylint fails.

File: travis/cfg/travis_run_pylint_pr.cfg (hunk 2 above)
Comment (moylop260) [inline travis/cfg/travis_run_pylint_pr.cfg]: You are right.
Thank you
I excluded in `pylint_exclude_61.cfg` but we need to exclude in `pylint_exclude_70.cfg` too.

Comment (moylop260) [inline travis/cfg/travis_run_pylint_pr.cfg]: Fixed!
Exclude `attribute-deprecated` in `travis_run_pylint_exclude_70.cfg`

**Comment by moylop260** [comment]
ping

File: travis/cfg/travis_run_pylint_beta.cfg (hunk 1 above)
Comment (dreispt) [inline travis/cfg/travis_run_pylint_beta.cfg]: You have a point about < 8.0.
But api.one usage in 8.0 is legal and should not raise an error.

Comment (moylop260) [inline travis/cfg/travis_run_pylint_beta.cfg]: @dreispt 
Can you check [CONTRIBUTING.md#field](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#field)?
`@api.one method: For v8 is recommended use @api.multi and avoid use @api.one, for compatibility with v9 where is deprecated @api.one.`

Comment (moylop260) [inline travis/cfg/travis_run_pylint_beta.cfg]: A similar case with: [CONTRIBUTING.md#idioms](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms)
`Exceptions: Use from openerp.exceptions import Warning as UserError (v8) or from openerp.exceptions import UserError (v9) or find a more appropriate exception in openerp.exceptions.py`

**Comment by pedrobaeza** [comment]
@moylop260, can we make specific configurations for each version instead of only beta/master?

**Comment by moylop260** [comment]
@pedrobaeza 
I don't understand you
Can you review next documentation?
https://github.com/OCA/maintainer-quality-tools/blob/master/travis/test_pylint#L5-L43

**Comment by moylop260** [comment]
@pedrobaeza 
Or Can you help me what do you wait of this PR?
Note: This PR enable new checks (don't change the current process)

**Comment by pedrobaeza** [comment]
If we enable these new checks, they will be applied to all branches. That's why I ask if we can have separate configurations for each branch/version. To avoid these new checks in old versions (< 8.0)

**Comment by moylop260** [comment]
@pedrobaeza 
The configuration of this PR is:

| msg  | 9.0  | 8.0  | 7.0  | 6.1  | is beta msg?    |   PR or branches  |
|---|---|---|---|---|---|---|
|   api-one-deprecated  |   yes |   yes |   n/a |   n/a |   yes |   PR  |
|   Attribute-deprecated    |   yes |   yes |   no  |   no  |   yes |   PR  |
|   Javascript-lint |   yes |   yes |   yes |   n/a |   yes |   PR  |
|   license-allowed |   yes |   yes |   yes |   yes |   yes |   PR  |
|   method-required-super   |   yes |   yes |   yes |   yes |   yes |   PR  |

**Comment by pedrobaeza** [comment]
:+1:

**Comment by dreispt** [comment]
We don't want api-one-deprecated for 8.0.
And I dislike the idea of having new checks running on non-PRs <= 8.0 - it may turn many projects suddenly red.

**Comment by dreispt** [comment]
To be clear: unless a wide consensus otherwise, IMO new checks should be only in 9.0 and on PRs.

**Comment by moylop260** [comment]
@dreispt 
This change don't turn many projects suddenly red.

Please, Could you review the main table?
Checks enabled just in changes of pull request (no main branches).
And checks enabled as beta messages (without affects status build).

Some checks are valid in old versions, if you consider that a check should be inactivated in a old version you could tell us.

By example: attribute-deprecated (_defaults, _columns) is inactivated in v7 and v6.1

We need avoid `api.one` for 8.0 (as guidelines tell us) to minimize the migration progress to 9.0
apione

And following next recommendation [Recommend not using @api.one in v8](https://github.com/odoo/odoo/pull/8527)

Similar to guideline of `Warning`
warning
and we are enabled in v8 next check: [openerp-exception-warning](https://github.com/OCA/maintainer-quality-tools/blob/c7a14d85d85a591999ea4570ca5de93fec46f7b8/travis/cfg/travis_run_pylint_beta.cfg#L33)

**Comment by lmignon** [comment]
@moylop260 I'm against new check on <= 8.0 even on PR. We can't indefinitely ask to the contributors to apply new coding rules on each PR. By applying new rules on <=8.0, we will discourage new contributions and especially bug fixes. I'm OK with these new checks for 9.0. IMO, we can request correction of these new rules as a first step when porting a moduel from V8 to V9 (and these fixes should be applied on 8.0).
my 2¢ ...

**Comment by pedrobaeza** [comment]
The checks are not affecting build status, so I see no problem in appearing in case someone wants to fix them.

**Comment by lmignon** [comment]
@pedrobaeza Are you sure? For V8 new checks are listed in 'enable' not in 'enabled2beta' https://github.com/OCA/maintainer-quality-tools/pull/280/files#diff-6d39baeaa64a2c47fff0320a5d7f2b49L40

**Comment by pedrobaeza** [comment]
@moylop260 has said that in https://github.com/OCA/maintainer-quality-tools/pull/280#issuecomment-150391022, but I haven't checked. Let's wait what he answers

**Comment by dreispt** [comment]
@moylop260 On the other hand, we need the new checks on 9.0 ASAP. 
Can we have a `pylint_exclude_80.cfg` to leave 8.0 unaffected and merge this, and afterwards discuss about additional checks for 8.0?

**Comment by moylop260** [comment]
Please, review next table:
beta_msgs

Is beta msg? yes
Then the don't affect status build.

Please,
Review next section and tell me if don't is clear to change the description or logic:
https://github.com/OCA/maintainer-quality-tools/blob/master/travis/test_pylint#L5-L42

**Comment by lmignon** [comment]
@moylop260 Are you sure? For V8 new checks are listed in 'enable' not in 'enabled2beta' https://github.com/OCA/maintainer-quality-tools/pull/280/files#diff-6d39baeaa64a2c47fff0320a5d7f2b49L40 I just want to understand why what I read in your table seems to be different that what I understand from your code.

**Comment by moylop260** [comment]
A summary description of process of configuration files.

We have 2 files to enable messages (affecting build status):
 - `global.cfg` file to all branches. (Enabled in all branches)
 - `pr.cfg` file to each PR just in modules changed. (Enabled in PR)

We have 2 files to disable messages:
  - `beta.cfg` - Show previous errors enabled, but don't affect build status. (Enabled still but without changes in build)
  - `exclude_VERSION.cfg` - Don't show previous errors enabled just in VERSION (Disabled)

Then the process should be:
 - Beta process (First step):
    - Add your new check to beta and pr configuration files to show them and give feedback.
 - Stable process progressive (X months after):
    - Delete checks from beta messages configuration file to affect status build but just in PR's.
  - Stable process definitive (Y months after):
     - Move checks from pr.cfg to global.cfg, to affect status build in global branch, except in exclude version cases.

This pull request is in first step: `beta process`

**Comment by moylop260** [comment]
Maybe, We should add to documentation next comment https://github.com/OCA/maintainer-quality-tools/pull/280#issuecomment-150565683
Where is a good place to put it?

**Comment by moylop260** [comment]
Included from https://github.com/OCA/maintainer-quality-tools/commit/22a6a0bce39705942b112cd1e240ee00c1ecee9b

**Pull request closed without merging**